### PR TITLE
feat: migrate catalog timestamp summary to metric crate

### PR DESCRIPTION
**Pull request by tustvold**
This migrates the timestamp summary within the catalog to the new metric crate

### CODE REVIEW

File: tests/end_to_end_cases/metrics.rs [hunk 1]
```diff
@@ -33,7 +33,7 @@ pub async fn test_row_timestamp() {
     let db_name_attribute = format!("db_name=\"{}\"", db_name);
 
     // Should only be enabled for the system table
-    assert_eq!(lines.len(), 60);
+    assert_eq!(lines.len(), 61);
```
Comment (alamb) [inline tests/end_to_end_cases/metrics.rs:36]: why did it increase to 61?

File: server/src/db/catalog/chunk.rs [hunk 2]
```diff
@@ -242,7 +242,7 @@ pub struct ChunkMetrics {
     pub(super) memory_metrics: StorageGauge,
 
     /// Track ingested timestamps
-    pub(super) timestamp_histogram: TimestampHistogram,
+    pub(super) timestamp_histogram: Option<TimestampHistogram>,
```
Comment (alamb) [inline server/src/db/catalog/chunk.rs:245]: why did this need to become optional?

**Review by alamb — APPROVED** [review]

File: tests/end_to_end_cases/metrics.rs (hunk 1 above)
Comment (tustvold) [reply tests/end_to_end_cases/metrics.rs:36]: Prometheus histograms have to include +Inf. Previously it was actually incorrect :sweat_smile: It is now a legitimate histogram :tada:

File: server/src/db/catalog/chunk.rs (hunk 2 above)
Comment (tustvold) [reply server/src/db/catalog/chunk.rs:245]: I thought it was less confusing than the "optional" being internal to the thing called TimestampHistogram

**Review by crepererum — APPROVED** [review]

**Pull request merged**